vpr/src/route/rr_graph.cpp

```diff
@@ -305,6 +353,18 @@ static void load_uniform_connection_block_pattern(vtr::NdMatrix<int, 6>& tracks_
                                                   const int y_chan_width,
                                                   const enum e_directionality directionality);
 
+/**
+ * @brief Loads the tracks_connected_to_pin array with an unevenly distributed set of switches across the channel.
+ * 
+ *   @param tracks_connected_to_pin The funtion loads up this data structure with a track index for each pin ([0..num_pins-1][0..width-1][0..height-1][0..layer-1][0..sides-1][0..Fc-1]]).                        
+ *   @param pin_locations Physical pin informations, such as pin_index in the physical type, side, and etc.
+ *   @param x_chan_width Number of tracks in x-axis.
+ *   @param y_chan_width Number of tracks in y-axis.
+ *   @param Fc Actual Fc value described in the architecture file for all pins of the specific phyiscal type ([0..number_of_pins-1][0..number_of_segments-1]).
+ *   @param seg_index The segment index that the function is trying to connect to pins.
```

Comment:
seg_index --> this comment doesn't really tell me what this is.  Is it the index of the segment type to which we want to connect pins?